queries/go.scm

```diff
@@ -222,3 +222,87 @@
     .
   )
 ) @anonymousFunction @namedFunction
+
+;; keys in maps
+(literal_value
+  "{" @collectionKey.iteration.start.endOf
+  (keyed_element
+    (_) @collectionKey @collectionKey.trailing.start.endOf
+    ":"
+    (_) @collectionKey.trailing.end.startOf
+  ) @collectionKey.domain
+  "}" @collectionKey.iteration.end.startOf
+)
```

Comment:
I have mixed feelings about defining a scope’s pattern and its iteration scope’s pattern in one go. I think it's often clearer to define them separately. You could arguably define `collectionKey.iteration` and `@value.iteration` in one pattern. I think that might be clearer

That being said, it's more of a rule of thumb than a hard rule; if you feel defining `collectionKey` and `collectionKey.iteration` in the same pattern is clearer in this case, I don't have a strong objection

There will potentially be a slight performance implication later fwiw, because this pattern will yield the iteration scope once per scope instance in the iteration scope, and we silently merge them internally. Defining the pattern separately will only yield the iteration scope once